geode-core/src/main/java/org/apache/geode/admin/internal/ThreadMonitoringImpl.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.admin.internal;
+
+
+import java.util.Properties;
+import java.util.Timer;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.geode.admin.ThreadMonitoring;
+import org.apache.geode.distributed.internal.DistributionConfigImpl;
+import org.apache.geode.internal.i18n.LocalizedStrings;
+import org.apache.geode.internal.statistics.AbstractExecutorGroup;
+import org.apache.geode.internal.statistics.FunctionExecutionPooledExecutorGroup;
+import org.apache.geode.internal.statistics.GatewaySenderEventProcessorGroup;
+import org.apache.geode.internal.statistics.OneTaskOnlyExecutorGroup;
+import org.apache.geode.internal.statistics.PooledExecutorGroup;
+import org.apache.geode.internal.statistics.ScheduledThreadPoolExecutorWKAGroup;
+import org.apache.geode.internal.statistics.SerialQueuedExecutorGroup;
+
+public class ThreadMonitoringImpl implements ThreadMonitoring {
+
+  private static ConcurrentMap<Long, AbstractExecutorGroup> monitorMap;
+
+  /** Monitors the health of the entire distributed system */
+  private ThreadMonitoringProcess tmProcess = null;
+
+  static Properties nonDefault = new Properties();
```

Comment:
Please change all variables from static to non-static (static should only be constants) and make them all private. No other classes should be able to directly access state that is owned by this class.